### ISSUE TITLE
API available

### ISSUE THREAD

**Issue by whistmxl** [murkl/d2launcher#13]
Hello there,

whist here from the MXL dev team, I'm working on a new launcher to replace the one we have for windows currently. Thought you may want to know there's an API available to get the files for the latest mod release. The new windows launcher uses this API and you are free to integrate it in the Linux launcher you developed, it's a public API.

https://launcher.median-xl.com/docs

Once you fetch the release manifest from the API, the files can be downloaded @ `https://get.median-xl.com/get.php?type={channel}&tag={tag}&file={filename}` (these variables come from the manifest that was just fetched)

For example: `https://get.median-xl.com/get.php?type=mxl_release&tag=2.5.2&file=D2Sigma.dll`

Feel free to reach out via discord if you have questions/requests

**Comment by whistmxl** [comment]
That's fine, just know eventually the old endpoints will be decommissioned and no longer work, or at very least they may not have the latest version's files available, so I'm letting you know about the new API early so that you or someone from the open source community can update and hopefully avoid disruption with this tool.

**Comment by murkl** [comment]
Thanks for the reminder. Are the Diablo 2 1.13c files still necessary or does the new API include all the files needed? I see there are some new files available. Also glide wrapper and cnc-draw are included as default?

**Comment by murkl** [comment]
Is implemented in #16 
I've kept the Diablo II 1.13c files for legacy support (in case of patching an older patch from d2launcher version manager). Because the old patches (zip files in ~/.d2launcher/patches) does not include all the files available in the new API. First the old 1.13c files are extracted and then all the Median XL files (the new API files may override some files from 1.13c). This approach should work with the old and the new patch files. But I don't know if there any side effects with cnc-draw or glide wrapper since the new API provides the same files and will override existing files in diablo2 dir.

**Comment by murkl** [comment]
Closed because released in 4.0.0

**Comment by whistmxl** [comment]
Yes, the approach I'm taking with the new windows launcher essentially is to provide all the necessary files except the base MPQ packages, in order to keep everything legal and still keep the requirement of "you must purchase a legal copy of the game to install and run Median XL".

And yes, sven's glide wrapper and cnc-ddraw are shipped as part of the standard installation. Eventually sven's glide wrapper may or may not be replaced with a newer/better implementation, such as D2DX or D2GL, not sure yet. If these third party wrappers are an issue for unix users, let me know, I may be able to provide an API channel specifically for this that would exclude the third party wrappers.

**Comment by Thorinori** [comment]
So looks like we are going to need to update slightly for future patches. As of 2.8.2 of MXL the MPQs are now spread across a variety of medianxl-*.mpq files instead of one big MXL.mpq file, and future updates may not necessarily update each file. whist recommended removing old medianxl mpqs with each update.

My initial thought is just add in `rm medianxl-*` before extracting the patches, however that doesn't factor in MXL.mpq or backwards compatibility. Do you want to handle this @murkl since I would assume you have a better idea of how you would want to maintain backwards compatibility, or want me to just PR that rm call in?

**Comment by murkl** [comment]
Oh I see. We can use zipinfo to list all the files from the current patched file and remove this files from d2 dir before patching the new version. An error only occurs if the user has manually deleted the current patch file from patches dir.
In addition we can create a simple text file with all files coming from patch and use this file for future patches (files to delete before patching). I guess, this should supports the older patch files.

New properties for custom glide and cnc wrapper seems a nice feature. I will probably implement all this at the weekend.

**Comment by murkl** [comment]
Is implemented in 4.0.2

**Comment by murkl** [comment]
@Thorinori for excluding files you have to add this property: https://github.com/murkl/d2launcher?tab=readme-ov-file#exclude-files-from-update-patch

**Comment by Thorinori** [comment]
Very nice! That was quick, great job :D
